backend/danswer/connectors/linear/connector.py

```diff
@@ -0,0 +1,209 @@
+                            description
+                            descriptionData
+                            comments {
+                                nodes {
+                                    url
+                                    body
+                                }
+                            }
+                        }
+                    }
+                    pageInfo {
+                        hasNextPage
+                        endCursor
+                    }
+                }
+            }
+        """
+        )
+
+        has_more = True
+        endCursor = None
+        while has_more:
+            graphql_query = {
+                "query": query,
+                "variables": {
+                    "first": self.batch_size,
+                    "after": endCursor,
+                },
+            }
+            logger.debug(f"Requesting issues from Linear with query: {graphql_query}")
+
+            response = _make_query(graphql_query, self.linear_api_key)
+            response_json = response.json()
+            logger.debug(f"Raw response from Linear: {response_json}")
+            edges = response_json["data"]["issues"]["edges"]
+
+            documents: list[Document] = []
+            for edge in edges:
+                node = edge["node"]
+                documents.append(
+                    Document(
+                        id=node["id"],
+                        sections=[
+                            Section(
+                                link=node["url"],
+                                text=node["description"],
+                            )
+                        ]
+                        + [
+                            Section(
+                                link=node["url"],
+                                text=comment["body"],
+                            )
+                            for comment in node["comments"]["nodes"]
+                        ],
+                        source=DocumentSource.LINEAR,
+                        semantic_identifier=node["identifier"],
+                        metadata={
+                            "updated_at": node["updatedAt"],
+                            "team": node["team"]["name"],
+                        },
+                    )
+                )
+                yield documents
+
+            endCursor = response_json["data"]["issues"]["pageInfo"]["endCursor"]
+            has_more = response_json["data"]["issues"]["pageInfo"]["hasNextPage"]
+
+    def load_from_state(self) -> GenerateDocumentsOutput:
+        yield from self._process_issues()
+
+    def poll_source(
+        self, start: SecondsSinceUnixEpoch, end: SecondsSinceUnixEpoch
+    ) -> GenerateDocumentsOutput:
+        start_time = datetime.fromtimestamp(start, tz=timezone.utc)
+        end_time = datetime.fromtimestamp(end, tz=timezone.utc)
+
+        yield from self._process_issues(start_str=start_time, end_str=end_time)
+
+
+if __name__ == "__main__":
```

Comment:
remove this junk!

Reply:
after discussing, keeping in to allow for easier manual testing

Reply:
future connectors should have something like this + it should be added to all existing connectors